stdlib/bytes.mli

```diff
@@ -216,6 +216,39 @@ val escaped : bytes -> bytes
     @raise Invalid_argument if the result is longer than
     {!Sys.max_string_length} bytes. *)
 
+val escaped_ascii : bytes -> bytes
+(** [escape_ascii s] is [s] with all characters outside the printable US-ASCII
+    range represented by escape sequences.
+
+    All characters outside the US-ASCII printable range \[0x20 .. 0x7E\] are
+    escaped, as well as backslash (0x2F) and double-quote (0x22).
+
+    The function {!Scanf.unescaped} is a left inverse of [escaped],
+    i.e. [Scanf.unescaped (escaped s) = s] for any string [s] (unless
+    [escaped s] fails).
+
+    @raise Invalid_argument if the result is longer than
+    {!Sys.max_string_length} bytes.
+
+    @since 4.14.0 *)
+
+val escaped_utf8 : bytes -> bytes
+(** [escape s] is [s] with special characters replaced with escape sequences,
+    following the lexical conventions of OCaml.
+
+    The following characters are escaped:
+    double quote (['\"'], newline (['\n']), carriage return (['\r']),
+    tab (['\t']), backspace (['\b']), DEL (0x7F),
+    and everything in the \[0x00 .. 0x1F\] range.
+
+    @raise Invalid_argument if the result is longer than
+    {!Sys.max_string_length} bytes.
+
+    @since 4.14.0 *)
+
+(* Intentionally undocumented. *)
+val _escaped : ?escape_unicode:bool -> bytes -> bytes
```

Comment:
The name should not be `_name`, prefixing by `_` generally means unused functions. Also, you should move the function after `(**/**)` (at the same place as `unsafe_get`). This will remove the function from the documentation.

Reply:
Also `escape_unicode` is a very misleading label name, use `utf_8`.

Reply:
It was a part of the [plan](https://github.com/ocaml/ocaml/pull/10457#discussion_r654567517) that Xavier Leroy didn't expressly disapprove of, so I assumed it was an acceptable thing to do.

What should I name it if it's not acceptable? It should somehow be marked "do not use".
Duplication may be the cleanest even if not prettiest way to handle the issue.

Reply:
I would propose to either add an optional argument (`for_utf_8`?) to escaped , or add a single function `escaped_for: [ `Utf_8 | `Ascii] -> string -> string` to `Bytes` and `String` that can be exposed directly.

Reply:
@Octachron I expressed that idea earlier. I'm happy to change the implementation if we agree on it.
It will be compatible with all existing code if the new parameter is set to `false`, and indeed will avoid the issues with undocumented functions and naming disputes.

* Change the type of `Bytes.escaped` to `escaped : ?for_utf_8:bool -> bytes -> bytes`.
* Use it from the `String` module to provide `String.escaped : ?for_utf_8:bool -> bytes -> bytes`.
* Remove `escaped_ascii` and `escaped_unicode` from the implementation.

If no one objects, I'd like to also add an optional argument for excluding specific characters from escaping, to make it useful in different settings.

Reply:
Personally I'm still totally unconvinced by the usefulness of this PR. An this whether you add arguments for excluding specific characters or not: the escape syntax will remain the one of OCaml (and "%S" will likely correspond to `String.escape` forever). 

I think this API should rather be left in peace as it is now.